cmd/utils.go

```diff
@@ -85,3 +88,45 @@ func PrepareMigrationPrerequisites(sourceProfileString, targetProfileString, sou
 	}
 	return sourceProfile, targetProfile, ioHelper, dbName, nil
 }
+
+// MigrateData creates database and populates data in it.
+func MigrateData(ctx context.Context, targetProfile profiles.TargetProfile, sourceProfile profiles.SourceProfile, dbName string, ioHelper *utils.IOStreams, writeLimit int64, conv *internal.Conv, skipForeignKeys, schemaOnly, dataOnly bool) (*writer.BatchWriter, error) {
```

Comment:
This method signature mixes together entities which aren't on the same level. `skipForeignKeys` is a parameter of the schema and data commands, but is passed in together with `schemaOnly` and `dataOnly` booleans.

I will recommend passing in a key-value pair to this method with the `key` being the `command` name (schema, data etc.) and the value being an object which provides the user configurations.
That should enhance readability.

Reply:
updated to use interface structure